docs/mpi_functions.rst

```diff
@@ -0,0 +1,144 @@
+MPI Functions
+-------------
+
+|MPIFunction|_ is the solution for executing MPI applications remotely using Globus Compute.
+As an extension to |ShellFunction|_, the |MPIFunction|_ supports the same interface
+to specify the command to invoke on the endpoint as well as the capture of output
+streams. However, |MPIFunction|_ diverges from |ShellFunction|_ in the following ways:
+
+1. An |MPIFunction|_ requires a specification of the resources for its execution. This
+   specification must be set on the executor on the client-side. The ``resource_specification: dict``
+   takes the following options:
+
+   .. code-block:: python
+
+        executor.resource_specification = {
+            'num_nodes': <int>,        # Number of nodes required for the application instance
+            'ranks_per_node': <int>,   # Number of ranks / application elements to be launched per node
+            'num_ranks': <int>,        # Number of ranks in total
+        }
+
+2. |MPIFunction|_ is designed to be used with ``GlobusMPIEngine`` on the endpoint.
+   ``GlobusMPIEngine`` is required for the partitioning of a batch job (blocks) dynamically
+   based on the ``resource_specification`` of the |MPIFunction|_.
+
+
+3. |MPIFunction|_ automatically prefixes the supplied command with ``$PARSL_MPI_PREFIX``
+   which resolves to an appropriate mpi launcher prefix (for e.g, ``mpiexec -n 4 -host <NODE1,NODE2>``).
```

Comment:
If a user forgets to use `GlobusMPIEngine`, instead using (say) the default `GlobusComputeEngine`, what happens?

Reply:
With https://github.com/globus/globus-compute/pull/1610, an error